parser.go

```diff
@@ -40,6 +40,7 @@ func Parse(statement string) (*AST, error) {
 
 	if len(lexer.errors) != 0 {
 		lexer.ast.Errors = lexer.errors
+		return lexer.ast, lexer.errors[0]
```

Comment:
This is a change of behavior mentioned in the PR description. An error will always be returned by `Parser`, even if no syntax error is found. If the query has multiple statements with errors it returns the error of the first statement. If the client wants to know all errors, it needs to check `ast.Errors`.

Reply:
Nice, I think this is better.